hazelcast/src/main/java/com/hazelcast/util/executor/LoggingScheduledExecutor.java

```diff
@@ -50,28 +52,47 @@
  */
 public class LoggingScheduledExecutor extends ScheduledThreadPoolExecutor {
 
+    /**
+     * Default {@ling ScheduledThreadPoolExecutor#removeOnCancel} policy is not working when tasks are decorated.
+     */
+    private final boolean removeOnCancelPolicy;
     private final ILogger logger;
     private volatile boolean shutdownInitiated;
 
     public LoggingScheduledExecutor(ILogger logger, int corePoolSize, ThreadFactory threadFactory) {
+        this(logger, corePoolSize, threadFactory, (HazelcastProperties) null);
+    }
+
+    public LoggingScheduledExecutor(ILogger logger, int corePoolSize, ThreadFactory threadFactory,
+                                    HazelcastProperties properties) {
```

Comment:
minor: I would pass `boolean removeOnCancelPolicy` to the constructor, instead of `HazelcastProperties`.